scripts/gm/bin/lib-gm

```diff
@@ -0,0 +1,811 @@
+  fi
+}
+
+get_add_to_hermes() {
+  RESULT="$(stoml -q "$CONFIG_FILE" "${1}.add_to_hermes")"
+  if [ -z "$RESULT" ]; then
+    test "$GLOBAL_ADD_TO_HERMES" = "true"
+  else
+    test "$RESULT" = "true"
+  fi
+}
+
+get_node_id() {
+  GAIAD_BINARY="$(get_gaiad_binary "$1")"
+  HOME_DIR="$(get_home_dir "$1")"
+  # Todo: Shall we change this to is_configuration_valid?
+  if [ ! -f "${HOME_DIR}/config/node_key.json" ]; then
+    warn "Invalid configuration: no node key found for $1"
+    echo "DEADBEEFDEADBEEF"
+  else
+    "$GAIAD_BINARY" tendermint show-node-id --home "$HOME_DIR"
+  fi
+}
+
+get_wallet_mnemonic() {
+  RESULT="$(stoml -q "$CONFIG_FILE" "${1}.wallet_mnemonic")"
+  if [ -z "$RESULT" ]; then
+    echo "$GLOBAL_WALLET_MNEMONIC"
+  else
+    echo "$RESULT"
+  fi
+}
+
+get_wallet_hdpath() {
+  RESULT="$(stoml -q "$CONFIG_FILE" "${1}.wallet_hdpath")"
+  if [ -z "$RESULT" ]; then
+    echo "$GLOBAL_WALLET_HDPATH"
+  else
+    echo "$RESULT"
+  fi
+}
+
+create() {
+  i="$1"
+  echo "Creating $i config..."
+  GAIAD_BINARY="$(get_gaiad_binary "$i")"
+  HOME_DIR="$(get_home_dir "$i")"
+  HOME_ROOT="${HOME_DIR%%${i}}"
+  if a_in_b "$i" "$VALIDATORS"; then
+    # Removing '-' to work around https://github.com/cosmos/cosmos-sdk/issues/9373
+    NODE_DIR_PREFIX="$(echo "${i}v" | tr -d '-')"
+    EXEC_RESULT="$("$GAIAD_BINARY" testnet --chain-id "$i" --keyring-backend test --node-dir-prefix "$NODE_DIR_PREFIX" -o "$HOME_ROOT" --node-daemon-home . --v 1 2>&1)"
+    if [ "$EXEC_RESULT" != "Successfully initialized 1 node directories" ]; then
+      warn "could not create config for ${i}: \"$EXEC_RESULT\", skipping..."
+      return 1
+    else
+      mv "${HOME_ROOT}/${NODE_DIR_PREFIX}0" "${HOME_DIR}"
+      mv "${HOME_ROOT}/gentxs" "${HOME_DIR}"
+      WALLET_MNEMONIC="$(get_wallet_mnemonic "$i")"
+      WALLET_HDPATH="$(get_wallet_hdpath "$i")"
+      if [ -z "$WALLET_MNEMONIC" ]; then
+        if [ -z "$WALLET_HDPATH" ]; then
+          "$GAIAD_BINARY" keys add "wallet" --keyring-backend test --keyring-dir "${HOME_DIR}" --output json > "${HOME_DIR}/wallet_seed.json"
+        else
+          "$GAIAD_BINARY" keys add "wallet" --hd-path "$WALLET_HDPATH" --keyring-backend test --keyring-dir "${HOME_DIR}" --output json > "${HOME_DIR}/wallet_seed.json"
+        fi
+      else
+        if [ -z "$WALLET_HDPATH" ]; then
+          echo "$WALLET_MNEMONIC" | "$GAIAD_BINARY" keys add "wallet" --recover --keyring-backend test --keyring-dir "${HOME_DIR}" --output json > "${HOME_DIR}/wallet_seed.json"
+        else
+          echo "$WALLET_MNEMONIC" | "$GAIAD_BINARY" keys add "wallet" --hd-path "$WALLET_HDPATH" --recover --keyring-backend test --keyring-dir "${HOME_DIR}" --output json > "${HOME_DIR}/wallet_seed.json"
+        fi
+        sconfig "${HOME_DIR}/wallet_seed.json" -t string "mnemonic=${WALLET_MNEMONIC}" 1> /dev/null
+      fi
+      "$GAIAD_BINARY" add-genesis-account "wallet" "10000000stake,10000localcoin" --keyring-backend test --home "${HOME_DIR}"
```

Comment:
Instead of "localcoin" is it possible to parametrize the type of coins that get added to `wallet` in genesis? I'd ideally like to have the option to do "samoleans" but this is not a hard requirements (can get around it).

Reply:
Also, would be great to have `100_000_000` (or a larger number) instead of `10000` for localcoin.

Reply:
```suggestion
      "$GAIAD_BINARY" add-genesis-account "wallet" "10000000stake,100000000samoleans" --keyring-backend test --home "${HOME_DIR}"
```

We can raise the number easily and hard-coded samoleans are better than localcoin. We can also create an optional denom but let's leave that for a next release.

Now that the codebase reached 1000 lines, we might want to think of making this into a proper tool with Rust or Go...